digits/model/tasks/caffe_train.py

```diff
@@ -860,7 +860,10 @@ def _pycaffe_args(self, gpu_id):
             solver_type = solver_type_mapping[self.solver_type]
         except KeyError:
             raise ValueError("Unknown solver type {}.".format(self.solver_type))
-        gpu_script = "caffe.set_device({id});".format(id=gpu_id) if gpu_id else ""
+        if gpu_id:
```

Comment:
This looks a little suspect. If `gpu_id == 0`, then this will return False, even though GPU `0` is a valid choice.

How about `if gpu_id is not None:`? Even if it does the same thing, it's a little more clear what types of values you're expecting.